src/braket/analog_hamiltonian_simulator/rydberg/rydberg_simulator_helpers.py

```diff
@@ -55,33 +55,36 @@ def get_blockade_configurations(lattice: AtomArrangement, blockade_radius: float
 
     # The coordinates for atoms in the filled sites
     atoms_coordinates = np.array(lattice.sites)[np.where(lattice.filling)]
-    min_separation = 1e10  # The minimum separation between atoms, or filled sites
+    min_separation = float("inf")  # The minimum separation between atoms, or filled sites
     for i, atom_coord in enumerate(atoms_coordinates[:-1]):
         dists = np.linalg.norm(atom_coord - atoms_coordinates[i + 1 :], axis=1)
         min_separation = min(min_separation, min(dists))
 
-    configurations = [
-        "".join(item) for item in itertools.product(["g", "r"], repeat=sum(lattice.filling))
-    ]
+    bit_list_iter = itertools.product(["g", "r"], repeat=sum(lattice.filling))
+    configuration_iter = ("".join(item) for item in bit_list_iter)
 
     if blockade_radius < min_separation:  # no need to consider blockade approximation
-        return configurations
-    return [
-        config
-        for config in configurations
-        if validate_config(config, atoms_coordinates, blockade_radius)
-    ]
+        return {config: ind for ind, config in enumerate(configuration_iter)}
+
+    ind = 0
+    configurations = {}
+    for config in configuration_iter:
+        if validate_config(config, atoms_coordinates, blockade_radius):
+            configurations[config] = ind
+            ind += 1
+
+    return configurations
 
 
 def _get_interaction_dict(
-    program: Program, rydberg_interaction_coef: float, configurations: List[str]
+    program: Program, rydberg_interaction_coef: float, configurations: Dict[str, int]
 ) -> Dict[Tuple[int, int], float]:
     """Return the dict contains the Rydberg interaction strength for all configurations.
 
     Args:
         program (Program): An analog simulation program for Rydberg system with the interaction term
         rydberg_interaction_coef (float): The interaction coefficient
-        configurations (List[str]): The list of configurations that comply with the blockade
+        configurations (Dict[str, int]): The list of configurations that comply with the blockade
```

Comment:
While I'm not as worried about keeping the private functions backward compatible, can we also update the docs to reflect what's in the Dict?